app/modules/code_provider/github/github_service.py

```
@@ -506,22 +506,51 @@
+                                status_code=e.status,
+                                detail=f"GitHub API error: {str(e)}"
+                            )
+                    else:
+                        logger.error(
+                            f"Error fetching branches for repo {repo_name}: {str(e)}", 
+                            exc_info=True,
+                        )
+                        raise HTTPException(
+                            status_code=500,
+                            detail=f"Error fetching branches: {str(e)}"
+                        )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)